rotkehlchen/data_import/importers/binance.py

```diff
@@ -383,20 +383,14 @@ def process_entry(
             data: BinanceCsvRow,
     ) -> None:
         asset = data['Coin']
-        category = AssetMovementCategory.WITHDRAWAL if data['Operation'] == 'Withdraw' else AssetMovementCategory.DEPOSIT  # else clause also covers 'Buy Crypto' & 'Fiat Deposit'  # noqa: E501
-        asset_movement = AssetMovement(
+        importer.add_history_events(write_cursor, [AssetMovement(
             location=Location.BINANCE,
-            category=category,
-            address=None,
-            transaction_id=None,
-            timestamp=timestamp,
+            event_type=HistoryEventType.WITHDRAWAL if data['Operation'] == 'Withdraw' else HistoryEventType.DEPOSIT,  # else clause also covers 'Buy Crypto' & 'Fiat Deposit'  # noqa: E501
+            timestamp=ts_sec_to_ms(timestamp),
             asset=asset,
-            amount=abs(data['Change']),
-            fee=Fee(ZERO),
-            fee_asset=A_USD,
-            link=f'Imported from binance CSV file. Binance operation: {data["Operation"]}',
-        )
-        importer.add_asset_movement(write_cursor=write_cursor, asset_movement=asset_movement)
+            balance=Balance(abs(data['Change'])),
+            extra_data={'memo': f'Imported from binance CSV file. Binance operation: {data["Operation"]}'},  # noqa: E501
```

Comment:
Duplicates will be enforced via unique constraints on the database as proposed in https://github.com/rotki/rotki/issues/9035. If I understand correctly this will apply to all history events, and not be something that needs handled specially here. If there is a constraint failure, that event will be skipped / added to a list of events that the user will need to go over and check for false positives.